jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -728,12 +733,31 @@ protected void initializeDefaultsComplete()
     {
     }
 
+    /**
+     * <p>Enters the scope of the {@link Context}.</p>
+     * <p> Attempts to set the {@link Thread#setContextClassLoader(ClassLoader) thread's context classloader}
+     * to the {@link #getClassLoader() configured one} if a non-null one was set.</p>
+     *
+     * @param contextRequest the context's {@link Request}
+     * @return the previous {@link Thread#getContextClassLoader() thread's context classloader}
+     */
     protected ClassLoader enterScope(Request contextRequest)
     {
+        CURRENT_CONTEXT.set(_context);
+
         ClassLoader lastLoader = Thread.currentThread().getContextClassLoader();
-        __context.set(_context);
-        if (_classLoader != null)
-            Thread.currentThread().setContextClassLoader(_classLoader);
+        if (_classLoader != null && _classLoader != lastLoader)
+        {
+            try
+            {
+                Thread.currentThread().setContextClassLoader(_classLoader);
+            }
+            catch (Throwable x)
+            {
+                if (LOG.isDebugEnabled())
+                    LOG.debug("error setting a context classloader on thread {}", Thread.currentThread(), x);
```

Comment:
I know this has been merged, 
but I think we should make this a warn the first time it happens.